core/src/traits/block_storage_interface.rs

```diff
@@ -1,12 +1,17 @@
 use async_trait::async_trait;
+use solana_rpc_client_api::config::RpcBlockConfig;
 use solana_sdk::{commitment_config::CommitmentLevel, slot_history::Slot};
-use solana_transaction_status::UiConfirmedBlock;
-use std::sync::Arc;
+use std::{ops::Range, sync::Arc};
+use crate::structures::produced_block::ProducedBlock;
 
 #[async_trait]
 pub trait BlockStorageInterface: Send + Sync {
-    async fn save(&self, slot: Slot, block: UiConfirmedBlock, commitment: CommitmentLevel);
-    async fn get(&self, slot: Slot) -> Option<UiConfirmedBlock>;
+    // will save a block
+    async fn save(&self, slot: Slot, block: ProducedBlock, commitment: CommitmentLevel);
+    // will get a block
+    async fn get(&self, slot: Slot, config: RpcBlockConfig) -> Option<ProducedBlock>;
+    // will get range of slots that are stored in the storage
+    async fn get_range(&self) -> Range<Slot>;
```

Comment:
nit: method name could explicitly state that it's not about blocks but slots only..e.g. "get_slot_range"